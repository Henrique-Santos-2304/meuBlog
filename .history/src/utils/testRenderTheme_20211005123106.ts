import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
