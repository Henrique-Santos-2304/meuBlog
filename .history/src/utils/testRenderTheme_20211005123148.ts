import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
