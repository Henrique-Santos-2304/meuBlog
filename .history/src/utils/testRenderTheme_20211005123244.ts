import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
