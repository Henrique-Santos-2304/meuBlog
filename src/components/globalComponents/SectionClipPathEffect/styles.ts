import styled from "styled-components";
import { props } from ".";

export const CliPathContent = styled.div<props>`
  width: 100vw;
  padding: 7rem 1rem;
  background: white;
  margin-top: ${({ mt }) => (mt ? mt : 0)};
  margin-bottom: ${({ theme }) => theme.spacings.small};
  clip-path: polygon(0 0, 100% 15%, 100% 100%, 0 90%);
`;
