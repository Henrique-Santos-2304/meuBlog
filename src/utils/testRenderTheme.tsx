import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
