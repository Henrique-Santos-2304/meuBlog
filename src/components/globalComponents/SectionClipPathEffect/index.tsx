import * as S from "./styles";
type childrenProps = {
  children: React.ReactNode;
};
const SectionClipPathEffect = ({ children }: childrenProps) => (
  <S.CliPathContent>{children}</S.CliPathContent>
);

export default SectionClipPathEffect;
