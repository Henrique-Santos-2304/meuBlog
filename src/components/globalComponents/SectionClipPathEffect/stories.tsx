import { Story, Meta } from "@storybook/react";
import SectionClipPathEffect from ".";

export default {
  title: "SectionClipPathEffect",
  component: SectionClipPathEffect,
} as Meta;

export const Basic: Story = () => (
  <SectionClipPathEffect>
    <p>Section Clip Path</p>
  </SectionClipPathEffect>
);

export const Default: Story = () => (
  <SectionClipPathEffect>
    <p>Section Clip Path</p>
  </SectionClipPathEffect>
);
