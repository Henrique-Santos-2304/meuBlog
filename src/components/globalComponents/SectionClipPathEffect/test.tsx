import SectionClipPathEffect from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

describe("<SectionClipPathEffect />", () => {
  it("should render ", () => {
    const { container } = renderTheme(
      <SectionClipPathEffect>
        <p>Section Clip Path</p>
      </SectionClipPathEffect>
    );
    expect(container).toMatchSnapshot();
  });
  it("should not have margin ", () => {
    renderTheme(
      <SectionClipPathEffect>
        <p>Section Clip Path</p>
      </SectionClipPathEffect>
    );
    const effectClipPath = screen.getByTestId(/containerEffectClipPath/i);
    expect(effectClipPath).toHaveStyle({ "margin-top": 0 });
  });
  it("should  have margin ", () => {
    renderTheme(
      <SectionClipPathEffect mt="2rem">
        <p>Section Clip Path</p>
      </SectionClipPathEffect>
    );
    const effectClipPath = screen.getByTestId(/containerEffectClipPath/i);
    expect(effectClipPath).toHaveStyle({ "margin-top": "2rem" });
  });
});
