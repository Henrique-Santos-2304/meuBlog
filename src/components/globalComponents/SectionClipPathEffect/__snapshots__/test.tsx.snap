// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionClipPathEffect /> should render  1`] = `
.c0 {
  width: 100vw;
  padding: 7rem 1rem;
  background: white;
  margin-top: 0;
  margin-bottom: 2.4rem;
  -webkit-clip-path: polygon(0 0,100% 15%,100% 100%,0 90%);
  clip-path: polygon(0 0,100% 15%,100% 100%,0 90%);
}

<div>
  <div
    class="c0"
    data-testid="containerEffectClipPath"
  >
    <p>
      Section Clip Path
    </p>
  </div>
</div>
`;
